server/memstore.go

```diff
@@ -287,15 +291,40 @@ func (ms *memStore) Stop() {
 	ms.mu.Unlock()
 }
 
-type observableMemStore struct{}
+func (ms *memStore) incObsCount() {
+	ms.mu.Lock()
+	ms.obsCount++
+	ms.mu.Unlock()
+}
+
+func (ms *memStore) decObsCount() {
+	ms.mu.Lock()
+
+	if ms.obsCount == 0 {
+		ms.mu.RUnlock()
+		return
+	}
+
+	ms.obsCount--
+	ms.mu.Unlock()
+}
+
+type observableMemStore struct {
+	ms *memStore
+}
 
 func (ms *memStore) ObservableStore(_ string, _ *ObservableConfig) (ObservableStore, error) {
-	return &observableMemStore{}, nil
+	ms.incObsCount()
```

Comment:
if there's a better way to get observable count on memstore pls let me know, not really happy with how I did this but seemed to be the least intrusive

Reply:
We should probably keep references, when you delete a message set it should take out the observables. That happens on filestore where I did first pass of observables. Need to backport to memstore most likely.